### PR TITLE
update flatpak manifest

### PR DESCRIPTION
**Pull request by muelli**
this is in preparation for flathub proposal. I would like the flathub manifest to be as close to the upstream maintained manifest as possible. These proposed  changes achieve a closer relationship.

### CODE REVIEW

**Comment by ezickler** [comment]
I wrote the original version of the manifest and also sometimes experienced build failures with portaudio, but i hadn't figured out that it could be the parallel build. Seams better now. 

This was one of my first flatpak manifest, so i didn't know about the portal option for the file chooser. So feel free to change it to better solutions.

**Comment by muelli** [comment]
> I wrote the original version of the manifest and also sometimes experienced build failures with portaudio, but i hadn't figured out that it could be the parallel build. Seams better now.

cool :)

> 
> This was one of my first flatpak manifest, so i didn't know about the portal option for the file chooser. So feel free to change it to better solutions.

It depends on the code. If the native GtkFilechooser is used, it should be able to access all files on the host. Does the app use a custom file chooser?

**Comment by ezickler** [comment]
I tested this without the ` - "--filesystem=xdg-documents" ` an it works fine on my fedora gone desktop. So yes it seams that its the default gtk file chooser

**Comment by alyberty** [comment]
The  `"--filesystemy=xdg-documents"` is part of the flatpack [sandboxing system](http://docs.flatpak.org/en/latest/sandbox-permissions.html#filesystem-access).

With this configuration, it only allows access to the documents folder, or if a user opens a file via xdg-open, the file will be made available to the sandbox.

The problem here is that, if the file references an external pdf, the pdf is not made available to the sandbox. This breaks expected behavior.

I'm not sure, what's the best option here. Ideally the sandbox is as restrictive as possible, without interfering in the user experience. But with the whole pdf referencing, I don't see the option to have any restrictions (apart from system directories).

**Comment by muelli** [comment]
if the xournalpp files are not self-contained, i.e. opening a single file through the file chooser does not work, then the sandbox restrictions need to be lifted (as is currently done). Some other apps do indeed expose the full home directory of the user to the sandboxed app. That's not ideal, as you mention. But it does help to preserve the UX.

**Comment by muelli** [comment]
FTR: There is a flathub app now: https://github.com/flathub/com.github.xournalpp.xournalpp
upstream involvement is highly appreciated.

**Comment by LittleHuba** [comment]
> I tested this without the `- "--filesystem=xdg-documents"` an it works fine on my fedora gone desktop. So yes it seams that its the default gtk file chooser

Yes we are using the native GTK file chooser. Is this ready to be merged or are there changes pending?

**Comment by muelli** [comment]
> Yes we are using the native GTK file chooser.

right, but it seems as if xournalpp files can reference further files. In that case, those additional files do not come from the file chooser.

>  Is this ready to be merged or are there changes pending?

yeah, I think this file is good enough for now.

**Comment by LittleHuba** [comment]
> right, but it seems as if xournalpp files can reference further files. In that case, those additional files do not come from the file chooser.

Yes and this will never change as it is a behaviour that is at least included in our current file format. The new one will solve this partially but will still allow for external files as there are usecases where it is required to not pack them.

File: com.github.xournalpp.xournalpp.yaml [hunk 1]
```diff
@@ -90,11 +95,9 @@ modules:
```
Comment (LittleHuba) [inline com.github.xournalpp.xournalpp.yaml]: Build works with CMake: Take a look at [this script](https://github.com/xournalpp/xournalpp/blob/mac-ci-builds/mac-setup/build-sndfile.sh). Even though it builds for MacOS it does work flawlessly.

File: com.github.xournalpp.xournalpp.yaml [hunk 2]
```diff
@@ -14,6 +13,7 @@ finish-args:
   # Wayland access
   - "--socket=wayland"
   # Needs to save files locally
+  # Why are portals not used? Does the app re-implement the open dialogue?
```
Comment (LittleHuba) [inline com.github.xournalpp.xournalpp.yaml]: This should be changed to a description why we require this. E.g. Xournal++ supports references to external files within their fileformat, hence global file access is required.

File: com.github.xournalpp.xournalpp.yaml [hunk 3]
```diff
@@ -57,10 +57,14 @@ modules:
       - type: archive
         url: https://libzip.org/download/libzip-1.5.2.tar.xz
         sha256: b3de4d4bd49a01e0cab3507fc163f88e1651695b6b9cb25ad174dbe319d4a3b4
+
   - name: libportaudiocpp
     buildsystem: autotools
     config-opts:
       - "--enable-cxx" # compile with c++ headers!!!
```
Comment (LittleHuba) [inline com.github.xournalpp.xournalpp.yaml]: This comment is also not really descriptive. Something like "Xournal++ uses the C++ headers of PortAudio, so compile them" would be better.

**Comment by LittleHuba** [comment]
@Technius is this one obsolete now as you did the work over at the flathub repo?

**Comment by Technius** [comment]
Yes.

**Pull request closed without merging**